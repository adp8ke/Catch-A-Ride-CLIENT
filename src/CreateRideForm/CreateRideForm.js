import React from 'react';
import RidesService from '../services/RidesService/rides-driver-service';





export default class CreateRideForm1 extends React.Component {



  //Grabs input values on the form
  grabValues=()=>{
  
    let time=document.getElementById('timeinput').value+':00';
    let date=document.getElementById('dateinput').value;
    let starting=document.getElementById('starting').value;
    let destination=document.getElementById('destination').value;
    let description=document.getElementById('description').value;
    let capacity=document.getElementById('capacity').value;
    

    return {
      starting,
      destination,
      date,
      time,
      description,
      capacity
    };


  }
  //Takes values from input form and puts them into an api call to the server.
  SubmitForm=(e)=>{
    e.preventDefault();
    let body=this.grabValues();
    console.log(body);
  //  RidesService.postNewRide(body);
  }
  render() {
    return (
      <section>
        <h2>Create Ride</h2>
        <form className="" onSubmit={this.SubmitForm}>
          <label>Starting Location</label>
          <input placeholder="Enter Location You Leave From" id='starting'required></input>
          <label>Destination</label>
          <input placeholder="Enter Destination " id='destination'required></input>
          <label>Date:</label>
          <input type="date" id='dateinput'required></input>
          <label>Time</label>
          <input type="time" id='timeinput' defaultValue='12:00'required></input>
          <label>Type:</label>
          <label> Description</label>
          <input placeholder="Enter Details About vehicle and Individual needs for ride" id='description'required></input>
          <label># of Seats</label>
          <select id='capacity'required>
            <option value='1'>1</option>
            <option value='2'>2</option>
            <option value="3">3</option>
            <option value="4">4</option>
            <option value="5">5</option>
            <option value="6">6</option>
            <option value="7">7</option>
          </select>
          <button type='submit'>Share a Ride!</button>
        </form>
      </section>
    );
  }
}
