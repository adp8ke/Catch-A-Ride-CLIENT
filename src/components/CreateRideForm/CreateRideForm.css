.newRideForm {
  line-height: 2em;
  display: block;
  width: 90%;
  margin: 0 auto;
  background: #22554e;
  color: white;
  padding: 15px 0;
}

.rideLabel {
  margin: 0 10px;
}

.rideSpace {
  background: white;
}

.createRide {
  margin: 0 auto;
  background: white;
  border-radius: 3px;
  text-align: center;
}

@media only screen and (min-width: 700px) {
  .newRideForm {
    width: 50%;
  }
}