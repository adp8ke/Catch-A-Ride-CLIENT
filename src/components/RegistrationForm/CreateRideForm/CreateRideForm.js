import React from "react";





export default class CreateRideForm extends React.Component {


  checkDate(e){
    e.preventDefault();

  let tdate=document.getElementById('dateinput').value
   
  console.log(tdate)
  }
  checkTime(e){
    e.preventDefault();

    let time= document.getElementById('timeinput').value+':00';
    console.log(time)
  }
  render() {
    return (
      <section>
        <h2>Create Ride</h2>
        <form className="">
          <label>Starting Location</label>
          <input placeholder="Enter Location You Leave From"></input>
          <label>Destination</label>
          <input placeholder="Enter Destination "></input>
          <label>Date:</label>
          <input type="date" id='dateinput'></input>
          <label>Time</label>
          <input type="time" id='timeinput'></input>
          <label>Type:</label>
          <label> Description</label>
          <input placeholder="Enter Details About vehicle and Individual needs for ride"></input>
          <label># of Seats</label>
          <select>
            <option value='1'>1</option>
            <option value='2'>2</option>
            <option value="3">3</option>
            <option value="4">4</option>
            <option value="5">5</option>
            <option value="6">6</option>
            <option value="7">7</option>
          </select>
          <button onClick={this.checkTime}>Share a Ride!</button>
        </form>
      </section>
    );
  }
}
