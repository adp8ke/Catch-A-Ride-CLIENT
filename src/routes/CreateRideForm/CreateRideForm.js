import React from 'react';
import {Link} from 'react-router-dom';
import RidesService from '../../services/RidesService/rides-driver-service';
// import RideContext from '../../context/RideContext';
import './CreateRideForm.css';
import {Redirect} from 'react-router-dom';

export default class CreateRideForm1 extends React.Component {
  // static contextType = RideContext;

  state = {
    created: null
  }

  grabValues = () => {

    let time = document.getElementById('timeinput').value;
    let date = document.getElementById('today').value;
    let starting = document.getElementById('starting').value;
    let destination = document.getElementById('destination').value;
    let description = document.getElementById('description').value;
    let capacity = document.getElementById('capacity').value;
    return JSON.stringify({
      starting,
      destination,
      date,
      time,
      description,
      capacity
    });


  }

  SubmitForm = (e) => {
    e.preventDefault();
    let body = this.grabValues();
    RidesService.postNewRide(body)
      .then(res => this.setState({ created: true, ride_id: res.id }));

  }
  componentDidMount(){
    let today = new Date().toISOString().substr(0, 10);
    document.querySelector("#today").value = today;
  }

  render() {
    const {ride_id} = this.state;
    if (this.state.created===true){
      return <Redirect to='/user-rides'/>
    }
    return (
      <>
        <h2>Create Ride</h2>
        {this.state.created === true ? <div className='rideMessage'>Ride Created! <Link className='ride-link' to={`/rides/${ride_id}`}>Go to Ride</Link></div> : ''}
        <form className='newRideForm' onSubmit={this.SubmitForm}>
          <div>
            <label className='rideLabel createStart' htmlFor='starting'>Starting Point</label>
            <input placeholder="Enter Location You Leave From" id='starting' required></input>
          </div>
          <div>
            <label className='rideLabel createDest' htmlFor='destination'>Destination</label>
            <input placeholder="Enter Destination " id='destination' required></input>
          </div>
          <div>
            <label className='rideLabel createDate' htmlFor='today' required>Date</label>
            <input type="date" id='today'></input>
          </div>
          <div>
            <label className='rideLabel createTime' htmlFor='timeinput' required>Time</label>
            <input type="time" id='timeinput' defaultValue='12:00'></input>
          </div>
          <div>
            <label className='rideLabel createDesc' htmlFor='description'> Description</label>
            <input placeholder="Enter Details About vehicle and Individual needs for ride" id='description' required></input>
          </div>
          <div>
            <label className='rideLabel createSeat' htmlFor='capacity'># of Seats</label>
            <select className='rideSpace' id='capacity'>
              <option value='1'>1</option>
              <option value='2'>2</option>
              <option value="3">3</option>
              <option value="4">4</option>
              <option value="5">5</option>
              <option value="6">6</option>
              <option value="7">7</option>
            </select>
          </div>
          <button className='createRide' type='submit'>Share A Ride!</button>
        </form>
      </>
    );
  }
}
