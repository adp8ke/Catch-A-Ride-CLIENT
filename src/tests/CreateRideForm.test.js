import React from 'react';
import ReactDOM from 'react-dom';
import {MemoryRouter} from 'react-router-dom';
import CreateRideForm from '../components/CreateRideForm/CreateRideForm';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MemoryRouter><CreateRideForm /></MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});
