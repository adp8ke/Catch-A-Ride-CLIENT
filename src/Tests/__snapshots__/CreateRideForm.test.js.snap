// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <h2>
    Create Ride
  </h2>,
  <form
    className="newRideForm"
    onSubmit={[Function]}
  >
    <div>
      <label
        className="rideLabel createStart"
        htmlFor="starting"
      >
        Starting Point
      </label>
      <br />
      <input
        id="starting"
        placeholder="Enter Location You Leave From"
        required={true}
        type="text"
      />
    </div>
    <div>
      <label
        className="rideLabel createDest"
        htmlFor="destination"
      >
        Destination
      </label>
      <br />
      <input
        id="destination"
        placeholder="Enter Destination "
        required={true}
        type="text"
      />
    </div>
    <div>
      <label
        className="rideLabel createDate"
        htmlFor="today"
        required={true}
      >
        Date
      </label>
      <input
        defaultValue="2019-10-03"
        id="today"
        type="date"
      />
    </div>
    <div>
      <label
        className="rideLabel createTime"
        htmlFor="timeinput"
        required={true}
      >
        Time
      </label>
      <br />
      <input
        defaultValue="12:00"
        id="timeinput"
        type="time"
      />
    </div>
    <div>
      <label
        className="rideLabel createDesc"
        htmlFor="description"
      >
         Description
      </label>
      <br />
      <textarea
        className="description"
        id="description"
        placeholder="Enter Details About vehicle and Individual needs for ride"
        required={true}
      />
    </div>
    <div>
      <label
        className="rideLabel createSeat"
        htmlFor="capacity"
      >
        # of Seats
      </label>
      <br />
      <select
        className="rideSpace"
        id="capacity"
      >
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
        <option
          value="3"
        >
          3
        </option>
        <option
          value="4"
        >
          4
        </option>
        <option
          value="5"
        >
          5
        </option>
        <option
          value="6"
        >
          6
        </option>
        <option
          value="7"
        >
          7
        </option>
      </select>
    </div>
    <button
      className="createRide"
      type="submit"
    >
      Share A Ride!
    </button>
  </form>,
]
`;
